packages/web3-eth2-beaconchain/src/index.ts

```diff
@@ -0,0 +1,606 @@
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @param {[string]} id (Optional) Array of ids, Either a hex string encoded public key (with 0x prefix) or validator index
+     * @param {Status} status (Optional) Validator status specification https://hackmd.io/ofFJ5gOmQpu1jjHilHbdQQ
+     * @returns {Promise} Array of the validators specified with balance, status and index
+     */
+
+    async getValidators(
+        stateId: StateId,
+        requestArguments?: Partial<RequestArguments>,
+        id?: [string],
+        status?: [Status]
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                method: `states/${stateId}/validators`,
+                params: { id, status },
+                ethVersion: 2,
+            });
+        } catch (error) {
+            throw Error(`Error getting validators: ${error.message}`);
+        }
+    }
+
+    /**
+     * Get validator from state by id
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot, hex string encoded stateRoot with 0x prefix
+     * @param {string} validatorId Either hex string encoded public key or validator index
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @returns {Promise} A validator specificied by state and id or public key along with status and balance
+     */
+
+    async getValidatorById(
+        stateId: StateId,
+        validatorId: string,
+        requestArguments?: Partial<RequestArguments>
+    ): Promise<RpcStringResult> {
+        try {
+            return await this.provider.request({
+                ...requestArguments,
+                method: `states/${stateId}/validators/${validatorId}`,
+                ethVersion: 2,
+            });
+        } catch (error) {
+            throw Error(`Error getting validator by Id: ${error.message}`);
+        }
+    }
+
+    /**
+     * Get Validator states from state
+     * @param {StateId} stateId State identifier, can be "head", "genesis", "finalized", "justified", slot or hex string encoded stateRoot with 0x prefix
+     * @param {object} requestArguments (Optional) rpcOptions, providerOptions, and desired returnType rpcOptions, providerOptions, and desired returnType
+     * @param {[string]} id (Optional) Array of either hex string encoded public keys or validator index
+     * @returns {Promise} Returns filterable array of validators with their balance, status and index.
+     */
+
+    async getValidatorBalances(
+        stateId: StateId,
+        requestArguments?: Partial<RequestArguments>,
```

Comment:
`reqeustArguments` should be the last argument to follow the same pattern in `web3-eth` (this would need to be updated for every method)